### PR TITLE
Improve text Insecure File Provider Paths Setting

### PR DESCRIPTION
**Pull request by najibraihan**

### CODE REVIEW

File: MOBILE_CLIENT/ANDROID/_HIGH/APK_FILE_PROVIDER_PERMISSIVE_EXTERNAL_PATH/description.md [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 Android defines multiple paths types:
 
 ```xml
 
+<root-path name="name" path="path"/>
+```
+
+* Checking the documentation of the [FileProvider](https://developer.android.com/reference/androidx/core/content/FileProvider) , you will not find the `<root-path...>` among the available paths.
+This path although not documented is available and can be used to provide access to internal storage of the app along with `/data` and `sdcard`.
+This path grants access to protected parts of the app and of the device.
```
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_HIGH/APK_FILE_PROVIDER_PERMISSIVE_EXTERNAL_PATH/description.md:14]:
```suggestion
This path grants access to protected parts of the app and of the device and thus exposes the application filesystem.
```

File: MOBILE_CLIENT/ANDROID/_HIGH/APK_FILE_PROVIDER_PERMISSIVE_EXTERNAL_PATH/description.md [hunk 2]
```diff
@@ -54,4 +61,12 @@ Android defines multiple paths types:
...
-The application specifies a permissive `android.support.FILE_PROVIDER_PATHS`.
+In the example below , we observe the provider has the root folder configuration that allows us to access home directory (which also includes /data and /sdcard directory).
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<paths xmlns:android="http://schemas.android.com/apk/res/android">
+    <root-path name="root" path="/"/>
+</paths>
+```
+This misconfiguration can be chained with other vulnerabilities like `Intent Redirection` to steal sensitive data
+or `Overwriting arbitrary files` to achieve arbitrary code execution by Overwriting native libraries.
```
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_HIGH/APK_FILE_PROVIDER_PERMISSIVE_EXTERNAL_PATH/description.md:72]:
```suggestion
or `Remote Code Execution` by overwriting native libraries.
```

File: MOBILE_CLIENT/ANDROID/_HIGH/APK_FILE_PROVIDER_PERMISSIVE_EXTERNAL_PATH/recommendation.md [hunk 3]
```diff
@@ -1,6 +1,11 @@
...
-
-Prefer `external-files-path` or use specific folders:
+* Avoid permissive settings when using `external-path`, like `'.'`.
+* Avoid using `root-path`.
+* Avoid assign the root path to path attribute `path='/.'`in all paths types.
+* You should make sure that you're only sharing files that are necessary and appropriate to share.
+* Avoid sharing sensitive files or files that contain sensitive information.
+* Ensure that you use the `<grant-uri-permission>` tag to control access to the shared files.
+* Prefer `external-files-path`.
+* use specific folders for path attribute:
```
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_HIGH/APK_FILE_PROVIDER_PERMISSIVE_EXTERNAL_PATH/recommendation.md:8]: This part needs more human-friendly text.

**Review by 3asm — DISMISSED** [review]

File: MOBILE_CLIENT/ANDROID/_HIGH/APK_FILE_PROVIDER_PERMISSIVE_EXTERNAL_PATH/recommendation.md (hunk 3 above)
Comment (3asm) [reply MOBILE_CLIENT/ANDROID/_HIGH/APK_FILE_PROVIDER_PERMISSIVE_EXTERNAL_PATH/recommendation.md]: Comment not addressed.

File: MOBILE_CLIENT/ANDROID/_HIGH/APK_FILE_PROVIDER_PERMISSIVE_EXTERNAL_PATH/recommendation.md [hunk 4]
```diff
@@ -1,6 +1,15 @@
...
+By making your app more secure, you help preserve user trust and device integrity, so to protect your app from this 
+vulnerability, here are some recommendations:
+* Be cautious about what files you share and only share files that are necessary and appropriate.
+* Don't share sensitive files or files that contain sensitive information.
+* When using external-path, avoid using permissive settings like '.' as the path.
+* Avoid using `root-path`.
+* Don't assign the root path '/.' to the path attribute in any type of path.
+* Use the <grant-uri-permission> tag to control access to shared files.
+* Prefer using `external-files-path` path type.
+* use specific folders for path attribute:
```
Comment (amine3) [inline MOBILE_CLIENT/ANDROID/_HIGH/APK_FILE_PROVIDER_PERMISSIVE_EXTERNAL_PATH/recommendation.md:12]: use specific folders for path attributes

**Review by amine3 — CHANGES_REQUESTED** [review]

**Review by amine3 — APPROVED** [review]

**Pull request merged**